### ISSUE TITLE
Training and inference imgsz parameter 

### ISSUE THREAD

**Issue by pablomuo** [ultralytics/ultralytics#8632]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I want to train a model for drone detection which will be used with a camera with a resolution 1920x1080, and the training will be with 50.000 images obtained by the internet from different sources. As the inference will be done with images of that resolution, in in the training process do I have to indicate that resolution in the imgsz argument? Do I also have to indicate the resolution in the imgsz argument of the inference? Thank you!

**Comment by glenn-jocher** [comment]
@pablomuo hello! 👋 Great question regarding the `imgsz` parameter for training and inference with YOLOv8.

For training, it's not strictly necessary to use the full resolution (1920x1080) of your camera images. Instead, you can train with a smaller `imgsz` for efficiency, such as 640 or 1280. YOLOv8 is quite robust and can generalize well even when trained on lower resolutions. However, if computational resources allow, training closer to your target resolution can be beneficial.

```bash
yolo detect train data=your_dataset.yaml model=yolov8n.pt epochs=100 imgsz=1280
```

For inference, using the `imgsz` parameter that matches your camera's resolution (1920x1080) can help in detecting smaller objects, as it preserves more detail. If performance is a concern, you might experiment with slightly lower resolutions to find a good balance between speed and accuracy.

```bash
yolo detect predict model=best.pt source=your_video.mp4 imgsz=1920,1080
```

In summary, while it's not mandatory to match the training `imgsz` with your camera resolution, doing so for inference can be advantageous. Feel free to experiment with different sizes to see what works best for your specific use case. Happy detecting! 🚀

**Comment by pablomuo** [comment]
Perfect! Now I am training with a imgsz of 1600, but now in some epochs appears this warning: WARNING ⚠️ NMS time limit 1.300s exceeded. 

Is there any solution for that?

Thanks!!

**Comment by glenn-jocher** [comment]
@pablomuo hey there! 😊 It looks like you're pushing the limits with a large `imgsz` of 1600, which is awesome for maximizing detail! The warning you're seeing about NMS time limit being exceeded is YOLOv8's way of letting you know that the non-maximum suppression step is taking longer than expected. This can happen with high-resolution inputs because there are more detections to process.

A quick fix to try is to increase the NMS time limit slightly in your configuration file or reduce the number of detections before NMS by increasing the confidence threshold. Here's how you could adjust the confidence threshold for inference:

```bash
yolo detect predict model=best.pt source=your_video.mp4 conf=0.3
```

This increases the minimum confidence for detections to be considered, potentially reducing the workload for NMS. Keep experimenting to find the sweet spot for your setup. Happy training! 🚀
